tests/potential/test_potential.py

```diff
@@ -0,0 +1,294 @@
+        self.assertEqual(m['A','A']['mass'], None)
+
+        #test setting all key values at once
+        m['A','A'] = {'energy':1.0, 'mass':2.0}
+        self.assertEqual(m['A','A']['energy'], 1.0)
+        self.assertEqual(m['A','A']['mass'], 2.0)
+
+        #test setting key values partially with = operator
+        m['A','A'] = {'energy':1.5}
+        self.assertEqual(m['A','A']['energy'], 1.5)
+        self.assertEqual(m['A','A']['mass'], None)
+
+        m['A','A'] = {'mass':2.5}
+        self.assertEqual(m['A','A']['energy'], None)
+        self.assertEqual(m['A','A']['mass'], 2.5)
+
+        #test setting key values partially using update()
+        m['A','A'].update(energy=2.5)  #using keyword arg
+        self.assertEqual(m['A','A']['energy'], 2.5)
+        self.assertEqual(m['A','A']['mass'], 2.5)
+
+        m['A','A'].update({'mass':0.5})  #using dict
+        self.assertEqual(m['A','A']['energy'], 2.5)
+        self.assertEqual(m['A','A']['mass'], 0.5)
+
+        #test accessing key param values individually
+        m['A','A']['energy'] = 1.0
+        self.assertEqual(m['A','A']['energy'], 1.0)
+        self.assertEqual(m['A','A']['mass'], 0.5)
+
+        m['A','A']['mass'] = 2.0
+        self.assertEqual(m['A','A']['energy'], 1.0)
+        self.assertEqual(m['A','A']['mass'], 2.0)
+
+        #test reset and get via iteration
+        params = ('energy', 'mass')
+        for j in params:
+            m['A','A'][j] = 1.5
+        self.assertEqual(m['A','A']['energy'], 1.5)
+        self.assertEqual(m['A','A']['mass'], 1.5)
+
+    def test_accessor_pairs(self):
+        """Test get and set methods for various pairs"""
+        #test set and get for initialized default
+        m = potential.CoefficientMatrix(types=('A','B'), params=('energy','mass'),
+                                        default={'energy':0.0, 'mass':0.0})
+        self.assertEqual(m['A','B']['energy'], 0.0)
+        self.assertEqual(m['A','B']['mass'], 0.0)
+        self.assertEqual(m['A','A']['energy'], 0.0)
+        self.assertEqual(m['A','A']['mass'], 0.0)
+        self.assertEqual(m['B','B']['energy'], 0.0)
+        self.assertEqual(m['B','B']['mass'], 0.0)
+
+        #test reset and get manually
+        m['A','B'] = {'energy':1.0, 'mass':2.0}
+        self.assertEqual(m['A','B']['energy'], 1.0)
+        self.assertEqual(m['A','B']['mass'], 2.0)
+        self.assertEqual(m['A','A']['energy'], 0.0)
+        self.assertEqual(m['A','A']['mass'], 0.0)
+        self.assertEqual(m['B','B']['energy'], 0.0)
+        self.assertEqual(m['B','B']['mass'], 0.0)
+
+        m['A','A'] = {'energy':1.0, 'mass':2.0}
+        self.assertEqual(m['A','B']['energy'], 1.0)
+        self.assertEqual(m['A','B']['mass'], 2.0)
+        self.assertEqual(m['A','A']['energy'], 1.0)
+        self.assertEqual(m['A','A']['mass'], 2.0)
+        self.assertEqual(m['B','B']['energy'], 0.0)
+        self.assertEqual(m['B','B']['mass'], 0.0)
+
+        m['B','B'] = {'energy':1.0, 'mass':2.0}
```

Comment:
And here, could you do this by update using the keyword version, and again different values?